secant/Features/App/App.swift

```diff
@@ -43,6 +65,83 @@ extension AppReducer {
     )
     .debug()
 
+    private static let appReducer = AppReducer { state, action, environment in
+        switch action {
+            /// Checking presense of stored wallet in the keychain and presense of database files in documents directory.
+        case .checkWalletInitialisation:
+            var walletInitialisationState: InitialisationState = .uninitialized
+
+            let fileManager = FileManager()
```

Comment:
I think this would belong in the environment if I understand correctly.

Reply:
yes, exactly issue #220 but I reordered some lines, the TODO was at different place